src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -294,10 +355,100 @@ public override async ValueTask<IReadOnlyList<BulkStateItem>> GetBulkStateAsync(
                 Message = invokeRequest,
             };
 
-            return await this.MakeGrpcCallHandleError(
-                 options => client.InvokeServiceAsync(request, options),
-                 headers,
-                 cancellationToken);
+            var callOptions = new CallOptions(headers: headers ?? new Metadata(), cancellationToken: cancellationToken);
+
+            // add token for dapr api token based authentication
+            var daprApiToken = Environment.GetEnvironmentVariable("DAPR_API_TOKEN");
+
+            if (daprApiToken != null)
+            {
+                callOptions.Headers.Add("dapr-api-token", daprApiToken);
+            }
+
+            // Common Exception Handling logic can be added here for all calls.
+            var grpcCall = client.InvokeServiceAsync(request, callOptions);
+            return grpcCall;
+        }
+
+        private async Task<ServiceInvocationResponse<TRequest, TResponse>> MakeInvokeRequestAsyncWithResponse<TRequest, TResponse>(
+            ServiceInvocationRequest<TRequest> request,
+            HTTPExtension httpExtension,
+            CancellationToken cancellationToken = default)
+        {
+            
+            var invokeResponse = new ServiceInvocationResponse<TRequest, TResponse>(request);
+            Any serializedData = null;
+            if (request.Body != null)
+            {
+                serializedData = TypeConverters.ToAny(request.Body, this.jsonSerializerOptions);
+            }
+
+            try
+            {
+                var grpcCall = MakeInvokeRequestAsync(request.AppId, request.MethodName, serializedData, httpExtension, cancellationToken);
+
+                var response = await grpcCall.ResponseAsync;
+                var responseHeaders = await grpcCall.ResponseHeadersAsync;
+                var trailers = grpcCall.GetTrailers();
+                var grpcStatus = grpcCall.GetStatus();
+
+                var headers = grpcCall.ResponseHeadersAsync.Result.ToDictionary(kv => kv.Key, kv => kv.ValueBytes);
+
+                    invokeResponse.Body = response.Data.Value.IsEmpty ? default : TypeConverters.FromAny<TResponse>(response.Data, this.jsonSerializerOptions);
+                    invokeResponse.Headers = headers;
+                    invokeResponse.Trailers = grpcCall.GetTrailers().ToDictionary(kv => kv.Key, kv => kv.ValueBytes);
+
+                if (IsResponseFromHttpCallee(headers))
+                {
+                    invokeResponse.HttpStatusCode = (HttpStatusCode)System.Enum.Parse(typeof(HttpStatusCode), Encoding.UTF8.GetString(headers[daprHTTPStatusHeader], 0, headers[daprHTTPStatusHeader].Length));
+                    invokeResponse.ContentType = Constants.ContentTypeApplicationJson;
```

Comment:
Since `headers` is a dictionary, can you replace this pattern with:

```C#
if (headers.TryGetValue(daprHttpStatusHeader, out var httpStatus))
{
}
```

This lets you check for presence and bind the value to a variable in one line.